src/Validator/ConstraintsSetFactory/Field/FieldConstraintsSetFactoryInterface.php

```diff
@@ -0,0 +1,22 @@
+<?php
+
+declare(strict_types=1);
+
+namespace AssoConnect\ValidatorBundle\Validator\ConstraintsSetFactory\Field;
+
+use Doctrine\ORM\Mapping\ClassMetadataInfo;
+use Symfony\Component\Validator\Constraint;
+
+/**
+ * @phpstan-import-type FieldMapping from ClassMetadataInfo
+ */
+interface FieldConstraintsSetFactoryInterface
+{
+    public function supports(string $type): bool;
+
+    /**
+     * @param FieldMapping $fieldMapping
+     * @return Constraint[]
+     */
+    public function getConstraints(array $fieldMapping): array;
```

Comment:
`Factory` sounds unatural to me without a "`create`" method.  Is it more like a "provider"?

Reply:
Good catch, thanks